gateway/src/main/java/eu/arrowhead/core/gateway/thread/ProviderSideSocketThreadHandler.java

```diff
@@ -145,6 +171,7 @@ public void onMessage(final Message message) {
 			} else {
 				Assert.notNull(currentThread.getOutputStream(), "Output stream is null.");
 				final byte[] bytes = relayClient.getBytesFromMessage(message, consumerGatewayPublicKey);
+				communicationStarted = true;
```

Comment:
you should also update currentThread.lastInteractionTime here before the getBytesFromMessage()